include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -339,15 +334,43 @@ SECTIONS
 
 	__app_last_address_used = .;
 
-	/* Pad out application ram area to make MPU friendly */
-	SECTION_PROLOGUE(app_pad, (NOLOAD OPTIONAL),)
-	{
-		. = ALIGN(_app_data_align);
-	} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+	/* Align the end of the application memory
+	 * with MPU alignment requirement.
+	 */
+#ifdef CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
+	. = ALIGN( 1 << LOG2CEIL(__app_ram_size));
+#endif
+	. = ALIGN(_region_min_align);
 	__app_ram_end = .;
 	__app_ram_size = __app_ram_end - __app_ram_start;
 #endif /* CONFIG_APPLICATION_MEMORY */
 
+#include <app_data_alignment.ld>
+
+#if defined(CONFIG_APP_SHARED_MEM)
```

Comment:
Yes, there is. I want to move _all_ fixed regions in the beginning of SRAM.
(Fixed regions = regions over which we do not do dynamic MPU programming afterwards)
Currently this concerns Application Memory, but soon we might get Region to Execute from RAM and No-Cache region (see relevant PRs) :)

Reply:
To elaborate a bit more; all linker sections corresponding to "feature" regions, such as Application Memory are moved to the beginning of image RAM, so, Kernel and Shared App memory can be placed in a single, continuous area in SRAM. The latter is essential for ARMv8-M, so that we may have all dynamic MPU regions (such as Stack Guards, thread stacks, memory partitions) be programmed within a single area; this is going to simplify the design.